index.js

```diff
@@ -102,44 +112,23 @@ Blake2b.prototype.digest = function (enc) {
 // libsodium compat
 Blake2b.prototype.final = Blake2b.prototype.digest
 
-Blake2b.WASM = wasm && wasm.buffer
+Blake2b.WASM = wasm
 Blake2b.SUPPORTED = typeof WebAssembly !== 'undefined'
 
 Blake2b.ready = function (cb) {
   if (!cb) cb = noop
-  if (!wasm) return cb(new Error('WebAssembly not supported'))
-
-  // backwards compat, can be removed in a new major
-  var p = new Promise(function (reject, resolve) {
-    wasm.onload(function (err) {
-      if (err) resolve()
-      else reject()
-      cb(err)
-    })
-  })
-
-  return p
+  if (!wasmPromise) return cb(new Error('WebAssembly not supported'))
+  return wasmPromise.then(() => cb())
```

Comment:
don't you need to error handle this promise? ie forward to the cb on error

Reply:
Good point, let me just fix that!